tools/hidl/hidlcomp.cpp

```diff
@@ -5414,113 +5417,94 @@ void EspServInfo::write_factory_impl()
+            indent.append('\t');
+    }
 
-                outf("%saccessmap.setValue(\"%s\", %s);\n", indent.str(), currAccessName.str(), translateAuthLevel(currAccessLevel.str()));
-                currAccessName.clear();
-                currAccessLevel.clear();
-                nameComplete = false;
-                continue;
-            }
-            else if (rawServiceAccessList[i] == ':')
-            {
-                if (currAccessName.length()==0)
-                {
-                    currAccessName.setf("%sAccess", methodname);
-                    outf("\n//processaccesslist: defaulted current name to : %s\n", currAccessName.str());
-                }
+    bool reportInfo() const override { return false; }
+    bool reportDebug() const override { return false; }
 
-                nameComplete = true;
-                continue;
-            }
-            else if (rawServiceAccessList[i] == '"')
-                continue;
+    void preEntry(size_t termCount) const override
+    {
+        outf("%sMapStringTo<SecAccessFlags> accessmap;\n", indent.str());
+    }
+    void entry(const char* name, const char* level) const override
+    {
+        outf("%saccessmap.setValue(\"%s\", %s);\n", indent.str(), name, level);
+    }
 
-            if (!nameComplete)
-                currAccessName.append(rawServiceAccessList[i]);
-            else
-                currAccessLevel.append(rawServiceAccessList[i]);
-        }
+protected:
+    void reportError(const char* fmt, va_list& args) const override
+    {
+        reportSomething("\nERROR: ", fmt, args);
     }
-    else
+    void reportWarning(const char* fmt, va_list& args) const override
+    {
+        reportSomething("//WARNING: ", fmt, args);
+    }
+    void reportInfo(const char* fmt, va_list& args) const override
+    {
+        reportSomething("//INFO: ", fmt, args);
+    }
+    void reportDebug(const char* fmt, va_list& args) const override
     {
-        outf("\n%saccessmap.setValue(\"%sAccess\", %s);\n", indent.str(), methodname, "SecAccess_Read"); //This seems to be the default per seclib
+        reportSomething("//DEBUG: ", fmt, args);
+    }
+
+    inline void reportSomething(const char* prefix, const char* fmt, va_list& args) const
+    {
+        outs(prefix);
+        voutf(fmt, args);
+        outs("\n");
     }
+};
+
+void writeAccessMap(int indentLevel, EspServInfo& svci, const char* serviceName, const char* serviceFragment, EspMethodInfo& mthi)
+{
+    HidlAccessMapScopeMapper scopeMapper({"EsdlService", "EsdlMethod"});
+    HidlAccessMapLevelMapper levelMapper;
+    HidlAccessMapReporter    reporter(indentLevel);
+    HidlAccessMapGenerator   generator(scopeMapper, levelMapper, reporter);
+
+    generator.setVariable("service", serviceName);
+    generator.setVariable("method", mthi.getName());
+    generator.insertScope("EsdlService", serviceFragment);
+    generator.insertScope("EsdlMethod", mthi.getMetaString(FEATEACCESSATTRIBUTE, NULL));
```

Comment:
What is FEATEACCESSATTRIBUTE?

Reply:
It is a pre-existing definition for "auth_feature".